### PR TITLE
Update 2020_10_25_101244_update_permissions_to_appshell_v2.php

### PR DESCRIPTION
**Pull request by j-dohnalek**
Allow to add flexibility to projects that manage permission differently. 

**Change:**
Check if permission exists, else create it

### CODE REVIEW

**Comment by fulopattila122** [comment]
Thanks for the PR. I like the approach of making it safer. However, I would not like to change its behavior, ie. the scope of the migration is to **update** existing permissions, and not to **create**.

Migrations must be as static as possible, thus retroactively changing a migration's behavior is a practice I prefer to avoid.

In case you were using other permission names (which is perfectly fine) you had to change your codebase accordingly.
At the same moment, you opted out of this feature set (which is again, perfectly OK). Thus your use case is not covered by this migration, and you have to migrate your set of permissions yourself within your application (also perfectly valid).

Would I accept this PR, it would solve your particular case, but it would also alter the behavior of everyone else. It would create an inconsistent state across users, meaning some people may or may not have those permissions created. Also, those, who opted out of those permission names would also need to delete them again if they want to keep opted out.

File: src/resources/database/migrations/2020_10_25_101244_update_permissions_to_appshell_v2.php [hunk 1]
```diff
@@ -8,24 +8,32 @@
 class UpdatePermissionsToAppshellV2 extends Migration
 {
     private $permissionsToMigrate = [
-        'list propertyvalues' => 'list property values',
+        'list propertyvalues'   => 'list property values',
```
Comment (fulopattila122) [inline src/resources/database/migrations/2020_10_25_101244_update_permissions_to_appshell_v2.php:11]: Please revert those style changes, and make sure StyleCI passes

File: src/resources/database/migrations/2020_10_25_101244_update_permissions_to_appshell_v2.php [hunk 2]
```diff
@@ -8,24 +8,32 @@
...
 
     public function up()
     {
         foreach ($this->permissionsToMigrate as $old => $new) {
-            Permission::findByName($old)->update(['name' => $new]);
+            if(Permission::whereName($old)->exists()) {
+                Permission::findByName($old)->update(['name' => $new]);
+            } else {
+                Permission::findOrCreate($new);
+            }
```
Comment (fulopattila122) [inline src/resources/database/migrations/2020_10_25_101244_update_permissions_to_appshell_v2.php:25]:
```suggestion
            $permission = Permission::findByName($old);
            if(null !== $permission) {
                $permission->update(['name' => $new]);
            }
```

File: src/resources/database/migrations/2020_10_25_101244_update_permissions_to_appshell_v2.php [hunk 3]
```diff
@@ -8,24 +8,32 @@
...
 
     public function down()
     {
         foreach ($this->permissionsToMigrate as $old => $new) {
-            Permission::findByName($new)->update(['name' => $old]);
+            if(Permission::whereName($new)->exists()) {
+                Permission::findByName($new)->update(['name' => $old]);
+            } else {
+                Permission::findOrCreate($old);
+            }
```
Comment (fulopattila122) [inline src/resources/database/migrations/2020_10_25_101244_update_permissions_to_appshell_v2.php:36]:
```suggestion
            $permission = Permission::findByName($new);
            if (null !== $permission) {
                $permission->update(['name' => $old]);
            }
```

**Review by fulopattila122 — CHANGES_REQUESTED** [review]

**Comment by j-dohnalek** [comment]
> Thanks for the PR. I like the approach of making it safer. However, I would not like to change its behavior, ie. the scope of the migration is to **update** existing permissions, and not to **create**.
> 
> Migrations must be as static as possible, thus retroactively changing a migration's behavior is a practice I prefer to avoid.
> 
> In case you were using other permission names (which is perfectly fine) you had to change your codebase accordingly.
> At the same moment, you opted out of this feature set (which is again, perfectly OK). Thus your use case is not covered by this migration, and you have to migrate your set of permissions yourself within your application (also perfectly valid).
> 
> Would I accept this PR, it would solve your particular case, but it would also alter the behavior of everyone else. It would create an inconsistent state across users, meaning some people may or may not have those permissions created. Also, those, who opted out of those permission names would also need to delete them again if they want to keep opted out.

Sure, that is reasonable argument. Thank you for taking it step further.

**Comment by j-dohnalek** [comment]
I have accepted all of your changes

File: src/resources/database/migrations/2020_10_25_101244_update_permissions_to_appshell_v2.php (hunk 1 above)
Comment (j-dohnalek) [reply src/resources/database/migrations/2020_10_25_101244_update_permissions_to_appshell_v2.php]: Done

**Comment by fulopattila122** [comment]
Thanks for the improvement Jiri!

**Pull request merged**